eventuals/then.h

```diff
@@ -14,30 +14,25 @@ template <typename T, typename Arg>
 using ValueFromMaybeComposable = typename std::conditional_t<
     !HasValueFrom<T>::value,
     decltype(Eventual<T>()),
-    T>::template ValueFrom<Arg>;
+    T>::template ValueFrom<Arg, std::tuple<>>;
```

Comment:
I'd actually like to keep `ValueFromMaybeComposable` a generic and reusable template, so let's instead update it to take in `Errors` just like `ErrorsFromComposable` and `ErrorsFromComposed` does and pass those errors here. Then when we use it down in the `Then` implementation we can pass `std::tuple<>` there.

That way these can possibly be used other places and it's only the call to them by the `Then` code that is saying that the errors we'd propagate in here is `std::tuple<>`. Make sense?